src/test/java/com/jwierzb/parkingspaces/controller/DriverControllerTest.java

```diff
@@ -0,0 +1,370 @@
+                .andExpect(content().string("Transaction doesnt exists. " + "Vehicle registration number: " + reg.toUpperCase()));
+
+    }
+
+
+    @Test
+    @WithMockUser(username = "test_user", roles = {"DRIVER"})
+    public void parkingShouldBeStarted_ifUserHaveVehicleNotParked() throws Exception {
+
+        UserEntity user = UserEntity.builder().username("asdf").build();
+        Vehicle vehicle = Vehicle.builder().registrationNumber("asdf").user(user).build();
+        Transaction transaction = Transaction.builder().endTime(null).user(user).startTime(LocalDateTime.now()).vehicle(vehicle).build();
+
+
+        String reg = "asdf";
+
+        when(vehicleDao.findByUserAndRegistrationNumber(any(), any())).thenReturn(Optional.of(new Vehicle()));
+
+        //null end time means there is no ongoing transactoins with thta vehicle
+        when(transactionDao.existsByUserAndVehicleAndEndTimeIsNull(any(), any())).thenReturn(false);
+        when(transactionDao.save(any())).thenReturn(transaction);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(transaction);
+
+        mockMvc.perform(post("/driver/parking/"+reg+"/start"))
+                .andExpect(status().isCreated())
+                .andExpect(content().string(jsonInString));
+
+    }
+    @Test
+    @WithMockUser(username = "test_user", roles = {"DRIVER"})
+    public void parkingShoulNotdBeStarted_ifUserHaveVehicleParked() throws Exception {
+
+        Vehicle vehicle = Vehicle.builder().registrationNumber("asdf").user(new UserEntity()).build();
+
+        String reg = "asdf";
+
+        when(vehicleDao.findByUserAndRegistrationNumber(any(), any())).thenReturn(Optional.of(new Vehicle()));
+
+        //uesr have vehicle parked
+        when(transactionDao.existsByUserAndVehicleAndEndTimeIsNull(any(), any())).thenReturn(true);
+
+
+        mockMvc.perform(post("/driver/parking/"+reg+"/start"))
+                .andExpect(status().isConflict())
+                .andExpect(content().string("Vehicle is already parked. Vehicle reg number: " + reg.toUpperCase()));
+
+    }
+
+    @Test
+    @WithMockUser(username = "test_user", roles = {"DRIVER"})
+    public void parkingShoulBeStopped_ifUserHaveVehicleParked() throws Exception {
+
+        UserEntity user = UserEntity.builder().username("asdf").build();
+        Vehicle vehicle = Vehicle.builder().registrationNumber("asdf").user(user).build();
+        Transaction transaction = Transaction.builder().endTime(null).user(user).startTime(LocalDateTime.now()).vehicle(vehicle).build();
+
+        String reg = "asdf";
+
+        when(vehicleDao.findByUserAndRegistrationNumber(any(), any())).thenReturn(Optional.of(new Vehicle()));
+
+        //user have vehicle parked
+        when(transactionDao.findByUserAndVehicleAndEndTimeIsNull(any(), any())).thenReturn(Optional.of(transaction));
+        when(transactionDao.save(any())).thenReturn(transaction);
```

Comment:
dlaczego to wszystko jest testowane na poziomie mock mvc? czy nie da się tego przetestować bardziej jednostkowo?